server/routes/profile.js

```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const router = express.Router();
+const protect = require('../middleware/auth');
+const { 
+    createProfile, 
+    updateProfile, 
+    getProfile, 
+    getAllProfiles 
+} = require('../controllers/profile');
+
+
+router.route('/').post(createProfile);
+
+router.route('/:id').patch(protect, updateProfile);
+
+router.route('/:id').get(protect, getProfile);
+
+router.route('/').get(protect, getAllProfiles);
```

Comment:
The routes look much better after the changes @ezeilo-su recommended 💯 🚀 Additionally, 
- I think it'd be better to `protect` the route on `line 12` doesn't make sense to create a profile for a user that doesn't have an account, and,
- It might be better **not** to require authentication for getting all profiles as mocks show the profile listing can be accessed by public users.